graphql_compiler/compiler/emit_sql.py

```diff
@@ -239,19 +338,182 @@ def start_global_operations(self):
+        #   array_agg(InnerVertex.fold_output_column) AS fold_output
+        # FROM OuterVertex
+        # INNER JOIN ... <- INNER JOINs compiled during Unfold block
+        # ON ...
+        #          ...
+        # INNER JOIN InnerVertex <- INNER JOINs compiled during Unfold block
+        # ON ...
+        # GROUP BY OuterVertex.uuid
+
+        if self._in_fold:
+            raise AssertionError(u'Fold block {} entered while inside another '
+                                 u'fold block at current location {}.'
+                                 .format(fold_scope_location, self._current_location_info))
+        # begin the fold
+        self._in_fold = True
+
+        # 1. get fold metadata
+        # location of vertex that is folded on
+        self._fold_inner_vertex_location = fold_scope_location
+        # fold information: source tables and join clause information
+        inner_alias, outer_alias, join_descriptor = self.get_fold_info(fold_scope_location)
+        join_info = (join_descriptor, self._current_alias)
+
+        # 2. initialize fold object
+        self._fold = SQLFoldObject(inner_alias, outer_alias, join_info)
+
+        # 3. add output columns to fold object
+        outputs = self.get_fold_outputs(fold_scope_location, join_descriptor)
+
+        # 4. add group by clause to fold object
+        group_by = [self._fold.outer_vertex_alias.c[join_descriptor.from_column]]
+        self._fold.set_outputs(outputs)
+        self._fold.set_group_by(group_by)
+
+    def construct_fold_joins(self, edge):
+        """Use the edge descriptors to create the join clause between the tables in the fold."""
+        join_clause = self._fold.outer_vertex_alias
+        join_clause = sqlalchemy.join(
+            join_clause,
+            self._fold.inner_vertex_alias,
+            onclause=(self._fold.outer_vertex_alias.c[edge.from_column] ==
+                      self._fold.inner_vertex_alias.c[edge.to_column])
+        )
+        return join_clause
+
+    def construct_fold_subquery(self, subquery_from_clause):
+        """Combine all parts of the fold object to produce the complete fold subquery."""
+        return sqlalchemy.select(
+            self._fold.outputs
+        ).select_from(
+            subquery_from_clause
+        ).group_by(
+            *self._fold.group_by
+        ).alias(self._alias_generator.generate_subquery())
+
+    def unfold(self):
+        """Complete the execution of a Fold Block."""
+        edge, outer_vertex = self._fold.join_info
+        subquery_from_clause = self.construct_fold_joins(edge)
+        fold_subquery = self.construct_fold_subquery(subquery_from_clause)
+
+        subquery_key = (self._fold_inner_vertex_location.base_location.query_path,
+                        self._fold_inner_vertex_location.fold_path)
+        # replaces the table first placed in the dict during the MarkLocation
+        # following the open of the fold scope
```

Comment:
Can you elaborate on why we need to replace the table? Or maybe just make a note in the MarkLocation function that the table is temporary and will need to be updated